src/main/java/nz/ac/auckland/kafka/http/sink/request/ApiRequest.java

```diff
@@ -109,14 +111,16 @@ private void processResponse() {
         try {
             int statusCode = connection.getResponseCode();
             log.info("Response Status: {}", statusCode);
-            RetryIndicator retryIndicator = getResponseIndicator();
-            log.info("Retry Indicator: {}", retryIndicator);
-            if(retryIndicator == RetryIndicator.UNKNOWN
-                    && CALLBACK_API_DOWN_HTTP_STATUS_CODE.contains(statusCode)){
-                throw new ApiRequestErrorException("Unable to connect to callback API: "
-                        + " received status: " + statusCode, kafkaRecord);
-            } else if (retryIndicator.shouldRetry) {
-                throw new ApiResponseErrorException("Received response retry=true", kafkaRecord);
+            if(Response.Status.Family.familyOf(statusCode) != Response.Status.Family.SUCCESSFUL){
+                RetryIndicator retryIndicator = getResponseIndicator();
+                log.info("Retry Indicator: {}", retryIndicator);
+                if(retryIndicator == RetryIndicator.UNKNOWN
+                        && CALLBACK_API_DOWN_HTTP_STATUS_CODE.contains(statusCode)){
+                    throw new ApiRequestErrorException("Unable to connect to callback API: "
+                            + " received status: " + statusCode, kafkaRecord);
+                } else if (retryIndicator.shouldRetry) {
+                    throw new ApiResponseErrorException("Received response retry=true", kafkaRecord);
+                }
```

Comment:
I prefer early returns 😄 

```suggestion
            if (Response.Status.Family.familyOf(statusCode) == Response.Status.Family.SUCCESSFUL) {
              return
            }
            
            RetryIndicator retryIndicator = getResponseIndicator();
            log.info("Retry Indicator: {}", retryIndicator);
            if (retryIndicator == RetryIndicator.UNKNOWN
                    && CALLBACK_API_DOWN_HTTP_STATUS_CODE.contains(statusCode)){
                throw new ApiRequestErrorException("Unable to connect to callback API: "
                        + " received status: " + statusCode, kafkaRecord);
            } else if (retryIndicator.shouldRetry) {
                throw new ApiResponseErrorException("Received response retry=true", kafkaRecord);
            }
```